api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -353,6 +353,10 @@ message HttpConnectionManager {
     // this type of upgrade. If no filters are present, the filter chain for
     // HTTP connections will be used for this upgrade type.
     repeated HttpFilter filters = 2;
+    // Determines if upgrades are enabled or disabled by default. This can be
+    // overriden on a per-route basis with :ref:`cluster
+    // <envoy_api_field_route.RouteAction.upgrade_configs>`. Defaults to true.
+    google.protobuf.BoolValue enabled = 3;
```

Comment:
This might just be me, but I dont think I got the full semantics of the enabled booleans until I read the code below. I think it might be worth it to spell out more, a truth table might even be worthwhile:

| *HCM Upgrade Enabled* | *Route Upgrade Enabled* | *Upgrade Enabled* |
|-----------------------|-------------------------|-------------------|
| T (Default)           | T (Default)             | T                 |
| T (Default)           | F                       | F                 |
| F                     | T (Default)             | T                 |
| F                     | F                       | F                 |